aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,125 @@
+
+If you have some logic that can produce a validation error, add that error to the model state's collection of errors by calling ``AddModelStateError`` as demonstrated below. Doing so will return the error message to the client and the view will display it.
+
+.. literalinclude:: validation/sample/MoviesController.cs
+   :language: c#
+   :lines: 48-69   
+   :emphasize-lines: 52-59
+   :dedent: 8
+   
+After model binding and validation are complete, you may want to repeat parts of it. For example, a user may have entered text in a field expecting an integer. You can invoke model binding to update the values of the model by using the ``UpdateModel`` or ``TryUpdateModel`` methods. When using ``UpdateModel``, you must put your code in `try-catch` block. It will attempt to update the model, but if it fails, it will throw an ``InvalidOperationException`` you must catch. ``TryUpdateModel`` attempts to update the model, but comes with a built-in exception handling in case it fails. Both methods will attempt to update the model state with validation errors, and then your view will display those errors. There is a similar pair of methods for validation called ``ValidateModel`` and ``TryValidateModel``. They work in the same manner but attempt to perform validation instead. 
+
+MVC will continue validating fields until reaches the maximum number of errors (200). You can configure this number by inserting the following code into the ``ConfigureServices`` method in the ``Startup.cs`` file, like so:
+
+.. literalinclude:: validation/sample/Startup.cs
+   :language: c#
+   :lines: 40-41,53-56,66
+   :dedent: 8
+
+Custom validation
+-----------------
+Data annotations work for most validation needs. However, some validation rules are specific to your business, as they're not just generic data validation such as ensuring a field is required or that it conforms to a range of values. For these scenarios, custom validation attributes are a great solution. Creating your own custom validation attributes in MVC is easy. Just inherit from the ``ValidationAttribute``, and override the ``IsValid`` method. The ``IsValid`` method accepts two parameters, the first is an object named `value` and the second is a ``ValidationContext`` object named `validationContext`. `Value` refers to the actual value from the field that your custom validator is validating. The `validationContext` object contains a property named `ObjectInstance`, which contains the information you need to validate your data.
+
+In the following sample, a business rule that states that movies released before 1960 belong to the classic movie genre. The ``[ClassicMovie]`` attribute checks the genre against the release date. If it is released before 1960, genre the property must be categorized as a classic movie. The attribute accepts an integer parameter representing the year that you can use to validate data. You can capture the value of the parameter in the attribute's constructor, as shown here:
```

Comment:
This paragraph describes a validation opposite to what the code below enforces. Code confirms movies with `Genre.Classic` were released before the given date.